### ISSUE TITLE
Output format

### ISSUE THREAD

**Issue by evadogstar** [R-TEK/colr_pickr#2]
Very beautiful color picker. Unfortunately, it's not possible to choose output format. Can you return `e.details.color.hex` etc as possible values from lib?

Thanks

**Comment by evadogstar** [comment]
All your examples set value to css styles, but what if I need to store some format into database :-(
Looks like need to search for other color picker :-(

**Comment by R-TEK** [comment]
> Very beautiful color picker. Unfortunately, it's not possible to choose output format. Can you return `e.details.color.hex` etc as possible values from lib?
> 

This is useful and something ill add

**Comment by R-TEK** [comment]
> All your examples set value to css styles, but what if I need to store some format into database :-(
> Looks like need to search for other color picker :-(

What do you mean by this?

Like if you needed to send the color output to the database? If its this then yeah thats already possible. The example use the color picker to set css sytles, only as an example of how you can use it.

So to understand your need better could you expand?

**Comment by evadogstar** [comment]
It is all about array with output of different values (hex, hsla, rga etc.) I'm already solved my issue. I convert default hsla output to hex with external script, but better if it will be possible to choose output values from array in the next version.

p.s. Maybe you can add description to readme how to initialize everything with jQuery. I'm already met the problem that after html button code with event updated (from external ajax request, it is the same button, with the same id, but new one....) then need to add event again. With jQuery code it will be more simple. As I know jquery `.on` made this automatically and need to maintain less code.

**Comment by R-TEK** [comment]
> It is all about array with output of different values (hex, hsla, rga etc.) I'm already solved my issue. I convert default hsla output to hex with external script, but better if it will be possible to choose output values from array in the next version.

Yes I'll add that feature in soon, most of the work is already there I just have passed the options to choose those different formats, so should be a quick add.

> p.s. Maybe you can add description to readme how to initialize everything with jQuery. I'm already met the problem that after html button code with event updated (from external ajax request, it is the same button, with the same id, but new one....) then need to add event again. With jQuery code it will be more simple. As I know jquery `.on` made this automatically and need to maintain less code.

Not sure what there is to initialize with Jquery, unless you mean changing the color picker source code to be reduce/simplify code. If so, this picker wasn't created to be a JQuery component, it can be used with JQuery though and you can replace the event listeners with the JQuery equivalent, etc.

Also, I couldn't fully make sense of your issue, is it that you have to different buttons that that trigger the same request and you have to make a new button instance and create the same request again in your code? Maybe, you could expand on this so I can see if there are solution for it, or possible features that could be added to the color picker to overcome it.

Thanks

**Comment by evadogstar** [comment]
>  it can be used with JQuery though and you can replace the event listeners with the JQuery equivalent, etc.
I only suggest to add this jQuery equivalent to README page :+1: 

> is it that you have to different buttons that that trigger the same request and you have to make a new button instance and create the same request again in your code?

No. it's the same buttons. They have the same IDs. They are placed inside `<form>`. I update the code of `<form>` html from server side by request (buttons also updated), because of that they lose event listining binding. After this I must re-init everything related to color picker. I guess that with jQuery event `.on` equivalent it can work out of the box after html update, because jQuery will automatically re-bind everything, but I'm not sure.

**Comment by R-TEK** [comment]
> > it can be used with JQuery though and you can replace the event listeners with the JQuery equivalent, etc.
> 
> I only suggest to add this jQuery equivalent to README page 
> 
> > is it that you have to different buttons that that trigger the same request and you have to make a new button instance and create the same request again in your code?
> 
> No. it's the same buttons. They have the same IDs. They are placed inside `<form>`. I update the code of `<form>` html from server side by request (buttons also updated), because of that they lose event listining binding. After this I must re-init everything related to color picker. I guess that with jQuery event `.on` equivalent it can work out of the box after html update, because jQuery will automatically re-bind everything, but I'm not sure. It's not a big issue. More like thinking out loud.
> 
> Thanks!

Oh right I understand now. The the color picker saves the button as a node, so even if a button has the same ID after being updated the instance will not update to that button, which seems to be the issue your having. I'll have to do some testing to see if there is a way to solve this (ill check how the .on works too), but for now a re-init seems the be the only way around. Anyway, good insight to use cases for this, so Ill look into it.

Thanks

**Comment by evadogstar** [comment]
Thanks! You can also try to use it with zubr foundation (ccs+js from cdn). It also have 2 issues:
1) It's break little bit input field of hex (move it to the left). Not a problem really.
2) If used with zubr foundation modals, it opened under it. I solve this by closing and then reopening foundation dialog. Same. Not a problem. 
I wrote it more for information. Really, it's not a problems.

**Comment by R-TEK** [comment]
Updated with the optional color output formats

**Comment by evadogstar** [comment]
@R-TEK ot works, need only to update readme and maybe add not only "hexa", but "hex" to object
thank you for update!

**Comment by R-TEK** [comment]
@evadogstar 
Yes, I'll add the new information to the readme.

You said add a hex value too, rather than just a hexa. What is you reason for wanting this, just curious because ultimately, that value will provide the same color even in hexa format?
Either way I think I will add this, as well as a rgb and hsl (along side the existing rgba and hsla)
